modules/dataframe/src/util/dataframeutil.cpp

```diff
@@ -524,6 +524,64 @@ std::shared_ptr<DataFrame> combineDataFrames(std::vector<std::shared_ptr<DataFra
     return newDataFrame;
 }
 
+#include <warn/push>
+#include <warn/ignore/conversion>
+std::vector<std::uint32_t> filteredRows(const Column& col, dataframefilters::Filters filters) {
+
+    if (col.getColumnType() == ColumnType::Categorical) {
+        const auto& catCol = dynamic_cast<const CategoricalColumn&>(col);
+        std::vector<std::uint32_t> rows;
+        for (auto&& [row, value] : util::enumerate<std::uint32_t>(catCol.getValues())) {
```

Comment:
humm getValues returns a new vector of strings that is very expensive :/ need a better way to iterate.

Reply:
should add a transformiterator (https://github.com/inviwo/inviwo/blob/master/include/inviwo/core/util/transformiterator.h) to CategoricalColumn that maps from index to string... that could return refs to the lookup table, so no copies should be needed at all...